src/main/java/com/github/housepower/jdbc/buffer/ByteArrayWriter.java

```diff
@@ -0,0 +1,63 @@
+package com.github.housepower.jdbc.buffer;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ *
+ */
+public class ByteArrayWriter implements BuffedWriter{
+    ByteBuffer buffer;
+
+    public ByteArrayWriter(ByteBuffer buffer) {
+        this.buffer = buffer;
+    }
+
+    public ByteArrayWriter(int size) {
+        this.buffer = ByteBuffer.allocate(size);
+    }
+
+    @Override
+    public void writeBinary(byte byt) throws IOException {
+        buffer.put(byt);
+        flushToTarget(false);
+    }
+
+    @Override
+    public void writeBinary(byte[] bytes) throws IOException {
+        writeBinary(bytes, 0, bytes.length);
+    }
+
+    @Override
+    public void writeBinary(byte[] bytes, int offset, int length) throws IOException {
+        for (int i = offset, max = offset + length; i < max; ) {
```

Comment:
you can try [this](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#put-byte:A-int-int-), limit length by remaining() and flushToTarget(), then put the rest etc. should be faster.

Reply:
Yes, I already have this it at [compressWriter](https://github.com/housepower/ClickHouse-Native-JDBC/blob/fdd039a46493d52a81843184d6477bd085b629d3/src/main/java/com/github/housepower/jdbc/buffer/CompressedBuffedWriter.java#L39) , and I will improve here,  thanks.